config/configgrpc/configgrpc_test.go

```diff
@@ -963,6 +963,118 @@ func TestClientInfoInterceptors(t *testing.T) {
 	}
 }
 
+func TestClientMetadataWithAuthInterceptorsAndIncludeMetadata(t *testing.T) {
+	testCases := []struct {
+		desc            string
+		tester          func(context.Context, ptraceotlp.GRPCClient)
+		includeMetadata bool
+	}{
+		{
+			desc: "overwritten",
+			tester: func(ctx context.Context, cl ptraceotlp.GRPCClient) {
+				resp, errResp := cl.Export(ctx, ptraceotlp.NewExportRequest())
+				require.NoError(t, errResp)
+				require.NotNil(t, resp)
+			},
+			includeMetadata: true,
+		},
+		{
+			desc: "found",
+			tester: func(ctx context.Context, cl ptraceotlp.GRPCClient) {
+				resp, errResp := cl.Export(ctx, ptraceotlp.NewExportRequest())
+				require.NoError(t, errResp)
+				require.NotNil(t, resp)
+			},
+			includeMetadata: false,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			mock := &grpcTraceServer{}
+			var l net.Listener
+
+			// prepare the server
+			{
+				gss := &ServerConfig{
+					NetAddr: confignet.AddrConfig{
+						Endpoint:  "localhost:0",
+						Transport: confignet.TransportTypeTCP,
+					},
+					IncludeMetadata: tC.includeMetadata,
+				}
+				gss.Auth = &configauth.Authentication{
+					AuthenticatorID: mockID,
+				}
+				host := &mockHost{
+					ext: map[component.ID]component.Component{
+						mockID: auth.NewServer(auth.WithServerAuthenticate(func(ctx context.Context, headers map[string][]string) (context.Context, error) {
+							return client.NewContext(ctx, client.Info{
+								Metadata: client.NewMetadata(metadata.Pairs("some-key-set-in-auth", "some-value-set-in-auth")),
```

Comment:
I see what you mean now. Authenticators are supposed to only provide data via the AuthData, not Metadata. Metadata is only supposed to be set by the parts that deal with the connection with the client, such as gRPC and HTTP servers (like those in receivers). Your downstream consumer of this information would have to call AuthData.GetAttribute(...). 

https://github.com/open-telemetry/opentelemetry-collector/blob/3d0189df8a26a5d654b51080dcd07ae57d394091/client/client.go#L26-L30

Here's an implementation example:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5cc9e03d054728b8f97d9214aa58910aa493c502/extension/basicauthextension/extension.go#L105-L107

Reply:
Yeah, but consider the following:

- This doesnt happen for oltp/http
- If I try putting in the AuthData, the batch processor will drop it. See this: https://github.com/open-telemetry/opentelemetry-collector/pull/10002

I feel the behaviours should be similar for oltp/http and oltp/grpc. Wouldn't you agree?

Reply:
Can you add a test case here for otlp/http? There should be no difference between the receivers, and if the authenticator is overriding the metadata, none should see the original headers after the authenticator runs.

Reply:
Table depicting what happens to metadata set in Auth:

| Protocol/Config | Include Metadata = True | Include Metadata = False |
| ------ | ------ | ----- |
| GRPC | overriden | not overriden |
| HTTP | not overriden | not overriden |

Reply:
@jpkrohling does the above make sense?